app/models/concerns/framework_assessmentable.rb

```diff
@@ -150,9 +150,9 @@ def previous_assessment
 
   def previous_responses
     @previous_responses ||= begin
-      return {} unless previous_assessment
+      return {} unless prefill_source
```

Comment:
Could you give a little context for this change pls - seems unrelated to performance tuning?

Reply:
this was calling previous assessment every time regardless of memoization, meaning it would call it around 53 times (depending on the framework response count). I realised we already set this value on the record and save it as prefill source. If we call prefill source here we avoid all the extra queries

Reply:
Awesome, nice explanation 👍